### PR TITLE
Remove 'HN'/'HN1' from N-terminal modifications for proline

### LINKED ISSUE #141: N-ter modification does not match N-ter proline
pdb2gmx adds the H1 and H2 atom on n-terminal prolines. It is not detected as the N-ter modification.
From what I see, the H2 atom is added to close to the  CD one, leading to an extra edge between them. Note that VMD also sees a bond there.

The list of affected systems from my screening can be obtained with:

```
grep 'ERROR - unknown-input - vermouth.processors.canonicalize_modifications - Could not identify the modifications for residues' /coarse/jon/pdb_screen/*/LOG | grep PRO | grep H2
```

An example output:

```
    INFO - step - vermouth - Guessing the bonds.
    INFO - step - vermouth - Repairing the graph.
   DEBUG - performance - vermouth.processors.repair_graph - Doing MCS matching for residue GLU83
    INFO - missing-atom - vermouth.processors.repair_graph - Missing atom GLU83:OE1
   DEBUG - missing-atom - vermouth.processors.repair_graph - Adding 1
   DEBUG - performance - vermouth.processors.repair_graph - Doing MCS matching for residue GLU83
    INFO - missing-atom - vermouth.processors.repair_graph - Missing atom GLU83:OE1
   DEBUG - missing-atom - vermouth.processors.repair_graph - Adding 3
    INFO - step - vermouth - Dealing with modifications.
   ERROR - unknown-input - vermouth.processors.canonicalize_modifications - Could not identify the modifications for residues ['PRO5'], involving atoms ['3-H2']
Traceback (most recent call last):
  File "/coarse/jon/links/dev/vermouth-martinize/vermouth/processors/canonicalize_modifications.py", line 264, in fix_ptm
    identified = identify_ptms(residue, res_ptms, options)
  File "/coarse/jon/links/dev/vermouth-martinize/vermouth/processors/canonicalize_modifications.py", line 179, in identify_ptms
    raise KeyError('Could not identify PTM')
KeyError: 'Could not identify PTM'
Traceback (most recent call last):
  File "/coarse/jon/Envs/apricot/bin/martinize2", line 7, in <module>
    exec(compile(f.read(), __file__, 'exec'))
  File "/coarse/jon/links/dev/vermouth-martinize/bin/martinize2", line 514, in <module>
    entry()
  File "/coarse/jon/links/dev/vermouth-martinize/bin/martinize2", line 402, in entry
    write_canon=args.write_canon,
  File "/coarse/jon/links/dev/vermouth-martinize/bin/martinize2", line 114, in pdb_to_universal
    vermouth.CanonicalizeModifications().run_system(canonicalized)
  File "/coarse/jon/links/dev/vermouth-martinize/vermouth/processors/processor.py", line 37, in run_system
    mols.append(self.run_molecule(molecule))
  File "/coarse/jon/links/dev/vermouth-martinize/vermouth/processors/canonicalize_modifications.py", line 315, in run_molecule
    fix_ptm(molecule)
  File "/coarse/jon/links/dev/vermouth-martinize/vermouth/processors/canonicalize_modifications.py", line 264, in fix_ptm
    identified = identify_ptms(residue, res_ptms, options)
  File "/coarse/jon/links/dev/vermouth-martinize/vermouth/processors/canonicalize_modifications.py", line 179, in identify_ptms
    raise KeyError('Could not identify PTM')
KeyError: 'Could not identify PTM'
```

### PR DESCRIPTION
**Pull request by pckroon**
This is a WIP PR to see if we can make the Nter modification match both prolines and other residues.

Fixes #141 (?)

### CODE REVIEW

**Comment by pckroon** [comment]
This should fix most issues with N-terminal prolines. It probably still breaks for zwitterionic prolines, but that's for another day.
I also suspect that the mappings for N-termini are slightly off, since HN is no longer used.

File: doc/source/conf.py [hunk 1]
```diff
@@ -199,9 +199,9 @@
 # to the VF2 isomorphism module instead.
 # See https://github.com/networkx/networkx/issues/3239
 intersphinx_aliases = {
-    ('py:class', 'networkx.classes.graph.Graph'): ('py:class', 'networkx.Graph'),
+    #('py:class', 'networkx.classes.graph.Graph'): ('py:class', 'networkx.Graph'),
     #('py:class', 'networkx.algorithms.isomorphism.vf2userfunc.GraphMatcher'): ('py:class', 'networkx.isomorphism.GraphMatcher'),
-    ('py:class', 'networkx.algorithms.isomorphism.vf2userfunc.GraphMatcher'): ('py:module', 'networkx.algorithms.isomorphism.isomorphvf2'),
+    #('py:class', 'networkx.algorithms.isomorphism.vf2userfunc.GraphMatcher'): ('py:module','networkx.algorithms.isomorphism.isomorphvf2'),
```
Comment (fgrunewald) [inline doc/source/conf.py:204]: This I don't understand. What has it to do with the prolines?

File: vermouth/molecule.py [hunk 2]
```diff
@@ -403,10 +403,6 @@ def force_field(self):
     def atoms(self):
         """
         All atoms in this molecule. Alias for `nodes`.
-
-        See Also
-        --------
-        :meth:`networkx.Graph.nodes`
```
Comment (fgrunewald) [inline vermouth/molecule.py:409]: This is a lint?! Why not keep the the also?

**Review by fgrunewald — APPROVED** [review]
If this fixes most of the prolines, then that's good enough for me. Zwitterionic prolines are probably of less concern. Can we document this limitation in a new issue though? The mapping only becomes problematic, if there is an EN on the side chain. Otherwise the beads should relax to the target topology, so that's fine in my opinion.

File: vermouth/molecule.py (hunk 2 above)
Comment (pckroon) [reply vermouth/molecule.py:409]: Not lint, no. But it breaks the doc building. See the commit description of efe3909. The root issue is that Graph.nodes is in the networkx intersphinx inventory as a property (as it should be), but there's no way to reference those in napoleon/autodoc/...

File: doc/source/conf.py (hunk 1 above)
Comment (pckroon) [reply doc/source/conf.py:204]: It's related to efe3909, a bit of cleanup.

**Pull request merged**